tests/HydrotestControllerTest.cs

```diff
@@ -0,0 +1,228 @@
+        Assert.IsNotNull(result);
+        Assert.AreEqual(93, result.Count());
+    }
+
+    [TestMethod]
+    public async Task GetEntriesByBoreholeIdForInexistentId()
+    {
+        var response = await controller.GetAsync(94578122).ConfigureAwait(false);
+        IEnumerable<Hydrotest>? hydrotests = response;
+        Assert.IsNotNull(hydrotests);
+        Assert.AreEqual(0, hydrotests.Count());
+    }
+
+    [TestMethod]
+    public async Task GetEntriesByBoreholeId()
+    {
+        var response = await controller.GetAsync(1005438).ConfigureAwait(false);
+        IEnumerable<Hydrotest>? hydrotests = response;
+        Assert.IsNotNull(hydrotests);
+        Assert.AreEqual(1, hydrotests.Count());
+        var hydrotest = hydrotests.Single();
+
+        Assert.AreEqual(hydrotest.Id, 12000069);
+        Assert.AreEqual(hydrotest.Type, ObservationType.Hydrotest);
+        Assert.AreEqual(hydrotest.Duration, 4461.0113159967641);
+        Assert.AreEqual(hydrotest.FromDepthM, 4594.9352618199473);
+        Assert.AreEqual(hydrotest.ToDepthM, 4446.2958402672293);
+        Assert.AreEqual(hydrotest.FromDepthMasl, 2308.5906170646617);
+        Assert.AreEqual(hydrotest.ToDepthMasl, 3394.1579820854395);
+        Assert.AreEqual(hydrotest.CompletionFinished, false);
+        Assert.AreEqual(hydrotest.Comment, "Libero debitis impedit cumque sit dolorum dignissimos.");
+        Assert.AreEqual(hydrotest.ReliabilityId, 15203156);
+        Assert.AreEqual(hydrotest.TestKindId, 15203178);
+
+        // Assert hydrotestresult
+        Assert.AreEqual(hydrotest.HydrotestResults.Count, 10);
+        var testResult = hydrotest.HydrotestResults.Single(r => r.Id == 13000375);
+        Assert.AreEqual(testResult.ParameterId, 15203200);
+        Assert.AreEqual(testResult.Value, 4390.010689483961);
+        Assert.AreEqual(testResult.MaxValue, 3484.6578809254142);
+        Assert.AreEqual(testResult.MinValue, 3823.8737230477263);
+        Assert.AreEqual(testResult.HydrotestId, 12000069);
+    }
+
+    [TestMethod]
+    public async Task EditAsyncValidEntityUpdatesEntity()
+    {
+        var originalHydrotest = new Hydrotest
+        {
+            Id = 1,
+            Type = ObservationType.Hydrotest,
+            StartTime = new DateTime(2006, 8, 21, 11, 00, 00).ToUniversalTime(),
+            EndTime = new DateTime(2015, 11, 13, 14, 00, 00).ToUniversalTime(),
+            Duration = 7909,
+            FromDepthM = 67.789,
+            ToDepthM = 78.15634,
+            FromDepthMasl = 67.112,
+            ToDepthMasl = 78.0043,
+            CompletionFinished = true,
+            Comment = "Test comment",
+            BoreholeId = 1008104,
+            ReliabilityId = context.Codelists.Where(c => c.Schema == "observ101").Single(c => c.Geolcode == 4).Id,
+            TestKindId = context.Codelists.Where(c => c.Schema == "htest101").Single(c => c.Geolcode == 1).Id,
+        };
+
+        var updatedHydrotest = new Hydrotest
+        {
+            Id = 1,
```

Comment:
PP: Hier sollte ein Kommentar rein, der besagt weshalb das Setzen und Speichern der `Id` in diesem Fall funktioniert und nicht automatisch von der DB generiert wird.